core/src/main/java/org/apache/accumulo/core/file/blockfile/cache/LruBlockCache.java

```diff
@@ -728,4 +746,11 @@ private long memorySize() {
   public void shutdown() {
     this.scheduleThreadPool.shutdown();
   }
+
+  @Override
+  public Lock getLoadLock(String blockName) {
+    // Would rather use Guava Striped, but its @Beta
+    int index = (blockName.hashCode() & 0x7fffffff) % loadLocks.length;
```

Comment:
I'm not sure the point of doing the AND here. Why not just do the modulus?

Reply:
The reason for the and is to drop the sign bit.

Reply:
I changed the code to `Math.abs()` which is easier to read.  It may be a bit slower but I doubt that matters relative locking cost.  Also Math.abs preserves an extra bit of info.

Reply:
Okay, I think it's fine either way. I would have expected modulus to always be positive, but I guess Java doesn't normalize modulus that way, like Python does. I think the AND is fine if there's a brief comment mentioning that it's to make it positive.

Reply:
See [`Math.floorMod()`](https://docs.oracle.com/javase/8/docs/api/java/lang/Math.html#floorMod-int-int-)